packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,672 @@
+import type { Maybe, MessageOrCCLogEntry, ValueID } from "@zwave-js/core";
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	parseBitMask,
+	validatePayload,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { getEnumMemberName, pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import { MessagePriority } from "../message/Constants";
+import {
+	CCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	SetValueImplementation,
+	SET_VALUE,
+	throwMissingPropertyKey,
+	throwUnsupportedProperty,
+	throwUnsupportedPropertyKey,
+	throwWrongValueType,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+// @publicAPI
+export enum SubsystemType {
+	Audible = 0x01,
+	Visual = 0x02,
+}
+
+// @publicAPI
+export enum SubsystemState {
+	Off = 0x00,
+	On = 0xff,
+}
+
+function getSignalingStateValueId(
+	endpoint: number | undefined,
+	subsystemType: SubsystemType,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Barrier Operator"],
+		endpoint,
+		property: "signalingState",
+		propertyKey: subsystemType,
+	};
+}
+
+function getSignalingStateMetadata(
+	subsystemType: SubsystemType,
+): ValueMetadata {
+	return {
+		...ValueMetadata.ReadOnlyUInt8,
```

Comment:
What is the significance of this being read only? In zwavejs2mqtt it's not editable. However, I can call the set API and modify it successfully. Should it also be writable?

Here's what zwave-js-server shows me, for reference:
```json
    {
      "endpoint": 0,
      "commandClass": 102,
      "commandClassName": "Barrier Operator",
      "property": "signalingState",
      "propertyKey": 1,
      "propertyName": "signalingState",
      "propertyKeyName": "1",
      "ccVersion": 0,
      "metadata": {
        "type": "number",
        "readable": true,
        "writeable": false,
        "min": 0,
        "max": 255,
        "label": "Signaling State (Audible)",
        "states": {
          "0": "Off",
          "255": "On"
        }
      },
      "value": 255
    },
```

Reply:
🙈 yes that should be writable too